### PR TITLE
Config: Expose merged configuration for easier debugging

### PR DESCRIPTION
**Pull request by gmalouf**
## Summary

This PR also introduces endpoint to retrieve local config in json format, scoped to private/admin API. It additionally adds a print to the algo-out file at startup of a fully hydrated/merged version of the configuration.

## Test Plan

Added test for the new `GetConfig` endpoint.

### CODE REVIEW

**Review by algorandskiy — COMMENTED** [review]
GetConfig REST LGTM

**Review by algorandskiy — APPROVED** [review]

File: cmd/algod/main.go [hunk 1]
```diff
@@ -174,6 +175,14 @@ func run() int {
 		log.Fatalf("Cannot load config: %v", err)
 	}
 
+	// log is not setup yet
+	fmt.Printf("Config loaded from %s\n", absolutePath)
+	fmt.Println("Configuration after loading/defaults merge: ")
```
[the remark this answers is not in the document]
Comment (algorandskiy) [reply cmd/algod/main.go:180]: I think algod-out.log is actually better place so the config is shown without double escaping quotes.

[the remark this answers is not in the document]
Comment (gmalouf) [reply cmd/algod/main.go:180]: Yeah, I thought about this while running it locally. Showed Pavel what it looked like in algo-out - think that is a better spot.

[the remark this answers is not in the document]
Comment (gmalouf) [reply cmd/algod/main.go:180]: Interesting, looking at it I didn't think it was double-encoded. RE: tweaking and restarting - if I'm trying to debug this, I move my out.log to an out.log.backup name (already doing manual intervention).

**Pull request merged**